src/System.IO.Ports/src/System/IO/Ports/SerialStream.cs

```diff
@@ -1681,9 +1681,26 @@ internal bool ShutdownLoop
                 }
             }
 
+            /// <summary>
+            /// Call WaitForCommEvent (which is a thread function for a background thread)
+            /// within an exception handler, so that unhandled exceptions in WaitForCommEvent
+            /// don't cause process termination
+            /// </summary>
+            internal void SafelyWaitForCommEvent()
+            {
+                try
+                {
+                    WaitForCommEvent();
+                }
+                catch(Exception ex)
```

Comment:
Understood.  The flip side of that is if there are really exceptions we're unaware of that may emerge, that could signal something really bad that we're just ignoring.

Regarding "use Task.Run instead"... why not?  It could be waited on in the stream's Dispose to ensure that no work associated with the stream is still going after disposal and to propagate any exceptions that may have occurred?  Would that lead to anything bad, like a deadlock?